dashboard/app/models/script_level.rb

```diff
@@ -59,15 +59,21 @@ def self.add_script_levels(script, lesson, raw_script_levels, counters, new_suff
       raw_script_level.symbolize_keys!
 
       properties = raw_script_level.delete(:properties) || {}
-
-      levels = Level.add_levels(raw_script_level[:levels], script, new_suffix, editor_experiment)
-
-      if new_suffix && properties[:variants]
-        properties[:variants] = properties[:variants].map do |old_level_name, value|
-          ["#{old_level_name}_#{new_suffix}", value]
-        end.to_h
+      raw_levels = raw_script_level[:levels]
+
+      # variants are deprecated. when cloning a script, retain only the first
+      # active level in each script level, discarding any variants.
+      if new_suffix && (raw_levels.length > 1 || properties[:variants])
+        first_active_level = raw_levels.find do |raw_level|
+          variant = properties[:variants].try(:[], raw_level[:name])
+          !(variant && variant[:active] == false)
```

Comment:
I'm not entirely clear on this boolean expression - why is `variant` included? What's the case where we want to choose this `raw_level` but `variant` is nil in this expression?

Reply:
active is the default. if there is no variant block, then it means the level is active. the ScriptDSL comments try to explain this: https://github.com/code-dot-org/code-dot-org/blob/969fa9c4d0da4c829aedaf1e59889134448f450d/dashboard/app/dsl/script_dsl.rb#L225-L235

that implementation has several problems, including (1) the script DSL and `raw_script_level` formats have are subtle differences, and (2) in the script DSL format, `active` has a different default for variants with experiments vs those without. because we want to deprecate this feature, I thought it would be better to match the new code to the existing behavior rather than trying to improve it and rely instead on good test coverage. 

while we're thinking about it, one way we could start deprecating this feature is to **get rid of script level experiments**, which no one should be using any more. side benefits of eliminating those would be that problem (2) would go away and problem (1) would be less bad. anyone with progress in an inactive / experimental variant would still see it when they go back to that level, per this logic: https://github.com/code-dot-org/code-dot-org/blob/fe06ad9d218cb5d45d722d7f95fc9f3f1b3a8f90/dashboard/app/controllers/script_levels_controller.rb#L421-L426

Reply:
here is some debug output to help explain the first part of my previous comment. when cloning the first script level in the unit test, which comes from this snippet of DSL: https://github.com/code-dot-org/code-dot-org/blob/91c163fb87420539e34ddfad4d87514c1b0fb023/dashboard/test/fixtures/test-fixture-variants.script#L2-L5

the local variables at this point in the code are as follows:
```
raw_levels: [{:name=>"Level 1"}, {:name=>"Level 2"}]
properties: {:variants=>{"Level 2"=>{:active=>false}}}
```